pkg/blobstore/grpcservers/byte_stream_server_test.go

```diff
@@ -125,6 +127,91 @@ func TestByteStreamServer(t *testing.T) {
 		require.Equal(t, io.EOF, err)
 	})
 
+	t.Run("ReadZSTDCompression", func(t *testing.T) {
+		// Test reading with ZSTD compression.
+		originalData := []byte("This is a test message that should be compressed with ZSTD")
+		blobAccess.EXPECT().Get(
+			gomock.Any(),
+			digest.MustNewDigest("", remoteexecution.DigestFunction_SHA256, "8b2c3f8a9d0e1f2a3b4c5d6e7f8a9b0c1d2e3f4a5b6c7d8e9f0a1b2c3d4e5f61", 58),
+		).Return(buffer.NewValidatedBufferFromByteSlice(originalData))
+
+		req, err := client.Read(ctx, &bytestream.ReadRequest{
+			ResourceName: "compressed-blobs/zstd/8b2c3f8a9d0e1f2a3b4c5d6e7f8a9b0c1d2e3f4a5b6c7d8e9f0a1b2c3d4e5f61/58",
+		})
+		require.NoError(t, err)
+
+		// Collect all compressed data
+		var compressedData []byte
+		for {
+			response, err := req.Recv()
+			if err == io.EOF {
+				break
+			}
+			require.NoError(t, err)
+			compressedData = append(compressedData, response.Data...)
+		}
+
+		// Decompress and verify
+		decoder, err := zstd.NewReader(nil)
+		require.NoError(t, err)
+		defer decoder.Close()
+
+		decompressedData, err := decoder.DecodeAll(compressedData, nil)
+		require.NoError(t, err)
+		require.Equal(t, originalData, decompressedData)
+	})
+
+	t.Run("ReadZSTDCompressionLargeData", func(t *testing.T) {
+		// Test reading large data with ZSTD compression.
+		originalData := make([]byte, 100000)
+		for i := range originalData {
+			originalData[i] = byte(i % 256)
+		}
+
+		blobAccess.EXPECT().Get(
+			gomock.Any(),
+			digest.MustNewDigest("", remoteexecution.DigestFunction_SHA256, "a1b2c3d4e5f6a7b8c9d0e1f2a3b4c5d6e7f8a9b0c1d2e3f4a5b6c7d8e9f0a1b2", 100000),
+		).Return(buffer.NewValidatedBufferFromByteSlice(originalData))
+
+		req, err := client.Read(ctx, &bytestream.ReadRequest{
+			ResourceName: "compressed-blobs/zstd/a1b2c3d4e5f6a7b8c9d0e1f2a3b4c5d6e7f8a9b0c1d2e3f4a5b6c7d8e9f0a1b2/100000",
+		})
+		require.NoError(t, err)
+
+		// Collect all compressed data
```

Comment:
Minor nit: please end all sentences in comments with a full stop (period). Thanks!